test/webaudio/audioworklet_in_out_stereo.c

```diff
@@ -0,0 +1,169 @@
+#include <assert.h>
+#include <string.h>
+#include <stdio.h>
+
+#include <emscripten/em_js.h>
+#include <emscripten/webaudio.h>
+
+// Tests processing two stereo audio inputs being mixed to a single stereo audio
+// output in process() (by adding the inputs together).
+
+// This needs to be big enough for the stereo output, 2x inputs and the worker stack
+#define AUDIO_STACK_SIZE 4096
+
+// Helper for MEMORY64 to cast an audio context or type to a void*
+#define WA_2_VOIDP(ctx) ((void*) (intptr_t) ctx)
+// Helper for MEMORY64 to cast a void* to an audio context or type
+#define VOIDP_2_WA(ptr) ((EMSCRIPTEN_WEBAUDIO_T) (intptr_t) ptr)
+
+// Count the audio callbacks and return after 375 frames (1 second with the default 128 size)
+//
+// *** Remove this in your own code ***
+//
+volatile int audioProcessedCount = 0;
+bool playedAndMixed(double time, void* data) {
+  if (audioProcessedCount >= 375) {
+    emscripten_force_exit(0);
+    return false;
+  }
+  return true;
+}
+
+// ID to the beat and bass loops
+EMSCRIPTEN_WEBAUDIO_T beatID = 0;
+EMSCRIPTEN_WEBAUDIO_T bassID = 0;
+
+// Creates a MediaElementAudioSourceNode with the supplied URL (which is
+// registered as an internal audio object and the ID returned).
+EM_JS(EMSCRIPTEN_WEBAUDIO_T, createTrack, (EMSCRIPTEN_WEBAUDIO_T ctxID, const char* url, bool looping), {
+  var context = emscriptenGetAudioObject(ctxID);
+  if (context) {
+    var audio = document.createElement('audio');
```

Comment:
Why are these two JS functions needed?  Is this functionality not available though existing emscripten APIs? 

Are folks expected to create their own audio elements like this?

Reply:
Yep, there's no provision (that I know of) to create audio tracks.

In reality the audio will probably come in via other means (we decode and mix Opus files, for example, or generate effects).